src/main/java/uk/ac/ebi/eva/pipeline/jobs/steps/VepInputGeneratorStep.java

```diff
@@ -49,35 +56,35 @@
  * https://examples.javacodegeeks.com/core-java/io/fileinputstream/compress-a-file-in-gzip-format-in-java/
  * http://www.journaldev.com/966/java-gzip-example-compress-and-decompress-file-in-gzip-format-in-java
  */
-
 @Configuration
 @EnableBatchProcessing
-@Import(JobOptions.class)
+@Import({NonAnnotatedVariantsMongoReaderConfiguration.class, VepInputFlatFileWriterConfiguration.class})
 public class VepInputGeneratorStep {
 
     private static final Logger logger = LoggerFactory.getLogger(VepInputGeneratorStep.class);
 
-    public static final String FIND_VARIANTS_TO_ANNOTATE = "Find variants to annotate";
-
     @Autowired
-    private StepBuilderFactory stepBuilderFactory;
+    @Qualifier(NON_ANNOTATED_VARIANTS_READER)
+    private NonAnnotatedVariantsMongoReader reader;
 
     @Autowired
-    private JobOptions jobOptions;
+    @Qualifier(VEP_INPUT_WRITER)
+    private ItemStreamWriter<VariantWrapper> writer;
+
+    @Bean(GENERATE_VEP_INPUT_STEP)
+    public Step generateVepInputStep(StepBuilderFactory stepBuilderFactory, JobOptions jobOptions) {
+        logger.debug("Building '" + GENERATE_VEP_INPUT_STEP + "'");
+
+        ObjectMap pipelineOptions = jobOptions.getPipelineOptions();
+        boolean startIfcomplete = pipelineOptions.getBoolean(JobParametersNames.CONFIG_RESTARTABILITY_ALLOW);
+        int chunkSize = pipelineOptions.getInt(JobParametersNames.CONFIG_CHUNK_SIZE);
 
-    @Bean
-    @Qualifier("vepInputGeneratorStep")
-    public Step variantsAnnotGenerateInputBatchStep() throws Exception {
-        return stepBuilderFactory.get(FIND_VARIANTS_TO_ANNOTATE)
-                .<DBObject, VariantWrapper>chunk(
-                        jobOptions.getPipelineOptions().getInt(JobParametersNames.CONFIG_CHUNK_SIZE))
-                .reader(new NonAnnotatedVariantsMongoReader(jobOptions.getDbName(), 
-                                                            jobOptions.getDbCollectionsVariantsName(),
-                                                            jobOptions.getMongoConnection()))
+        return stepBuilderFactory.get(GENERATE_VEP_INPUT_STEP)
+                .<DBObject, VariantWrapper>chunk(chunkSize)
+                .reader(reader)
                 .processor(new AnnotationProcessor())
```

Comment:
Depends on...

- Needs  job parameters? -> Bean with appropriate scope
- Extends InitializationBean? -> Bean (Use class injection if there is only one of that type, if more, then use naming)
- None of the above... plain old new object.